File: Assignment1/lock.py
class Lock:
    DEFAULT_STATE : bool = False # is the lock locked(true) or unlocked(false)
    DEFAULT_DIAMETER : float = 2.16 # inches

    def __init__(self, diameter : float = DEFAULT_DIAMETER, state : bool = DEFAULT_STATE):
        self._diameter = diameter
        self._state = state
        self._needKey = False

    @property 
    def diameter(self) -> float:
        return self._diameter
    
    @property
    def state(self) -> bool:
        return self._state

    @state.setter
    def set_needKey(self, value : bool) -> None:
        if value == True:
            self._needKey = True
        if value == False:
            self._needKey = False

    
 
        

File: Assignment2/src/Lock.cpp
#include <stdexcept>
#include "Lock.h"

namespace lock
{

const bool Lock::DEFAULT_STATE = false;
const double Lock::DEFAULT_DIAMETER = 2.16;

Lock::Lock(double _diameter, bool _state)
    : diameter(_diameter), state(_state)
    {



}

double Lock::getDiameter() const
{
    return diameter;
}

bool Lock::getState() const
{
    return state;
}

bool Lock::getNeedKey() const
{
    return needKey;
}